app/v2/models/responses/hateoas/HateoasMovementMessageIdsResponse.scala

```diff
@@ -32,20 +33,31 @@ object HateoasMovementMessageIdsResponse extends HateoasResponse {
         "self"                    -> Json.obj("href" -> getMessagesUri(movementId, receivedSince, movementType)),
         movementType.movementType -> Json.obj("href" -> getMovementUri(movementId, movementType))
       ),
-      "messages" -> messageIds.map(
+      "messages" -> messageIds.map {
         message =>
-          Json.obj(
-            "_links" -> Json.obj(
-              "self"                    -> Json.obj("href" -> getMessageUri(movementId, message.id, movementType)),
-              movementType.movementType -> Json.obj("href" -> getMovementUri(movementId, movementType))
-            ),
-            "id"                              -> message.id.value,
-            s"${movementType.movementType}Id" -> movementId.value,
-            "received"                        -> message.received,
-            "type"                            -> message.messageType,
-            "status"                          -> message.status
+          val jsMessageLinksObject = Json.obj(
+            "self"                    -> Json.obj("href" -> getMessageUri(movementId, message.id, movementType)),
+            movementType.movementType -> Json.obj("href" -> getMovementUri(movementId, movementType))
           )
-      )
+          if (message.status == MessageStatus.Pending) {
+            Json.obj(
+              "_links"                          -> jsMessageLinksObject,
+              "id"                              -> message.id.value,
+              s"${movementType.movementType}Id" -> movementId.value,
+              "received"                        -> message.received,
+              "status"                          -> message.status
+            )
+          } else {
+            Json.obj(
+              "_links"                          -> jsMessageLinksObject,
+              "id"                              -> message.id.value,
+              s"${movementType.movementType}Id" -> movementId.value,
+              "received"                        -> message.received,
```

Comment:
Message type is optional right? We could prevent the duplicated code by doing something similar to what we do for body in the Hateos response for the movement response

```
  message.messageType .map(
         type => Json.obj("type" -> type)
       )
       .getOrElse(Json.obj())
```

Reply:
nope, Message type is not optional

Reply:
hmmm, that's gonna have to change in the future for reasons given above -- but for now it can probably be left as is because we've not hit the point where it _has_ to be optional and I don't want scope creep to accidentally occur.